src/main/java/com/neroreflex/plugins/Calc.java

```diff
@@ -18,65 +18,81 @@
 
 package com.neroreflex.plugins;
 
-import java.text.SimpleDateFormat;
-import java.util.Calendar;
+import java.io.IOException;
+import java.io.Writer;
 import java.util.Vector;
 import com.neroreflex.pizza.*;
-import java.io.BufferedReader;
-import java.io.IOException;
-import java.io.InputStreamReader;
-import java.net.Socket;
-import java.text.SimpleDateFormat;
-import java.util.Calendar;
-import java.util.TimeZone;
+import com.github.gianlucanitti.javaexpreval.*;
 
 /**
- * Un plugin per eseguire semplici operazioni matematiche
+ * Un plugin per risolvere espressioni matematiche
  *
  * @author Nitti Gianluca
  */
 public final class Calc extends Trancio {
+
+    private class IrcWriter extends Writer{
+        private String chan;
+        private String buffer;
+
+        public IrcWriter(String chan){
+            this.chan = chan;
+            buffer = "";
+        }
+
+        @Override
+        public void write(char[] cbuf, int off, int len) throws IOException {
+            for(int i = off; i < len; i++)
+                buffer += cbuf[i];
+        }
+
+        @Override
+        public void flush() throws IOException {}
+
+        @Override
+        public void close() throws IOException {
+            if(buffer.length() != 0)
+                for(String s: buffer.split("[\\r\\n]+"))
+                    sendMessage(new Message(chan, s));
+            buffer = "";
+        }
+    }
+
     protected String onHelp() {
-        return "<number1> <operation> <number2> - where operation can either be +, - * or /";
+        return "Use !calc <expression> to evaluate an expression; supports integer and decimal numbers, parenthesis and +,-,*,/,^ binary operators. !calc verbose <expression> shows the steps done.";
```

Comment:
Il bot aggiunge sempre !calc prima della stringa ricevuta dal plugin, quindi un help migliore sarebbe:
[<verbose>] <expr> - where expr is the expression to be evaluated and verbose, if present, forces the plugin to write resolution steps.